lib.nix

```diff
@@ -12,10 +12,23 @@ let
        then result
        else default;
 
+  cleanHaskellSource = src:
+    if (builtins.typeOf src) == "path"
+      then lib.cleanSourceWith {
+        filter = with pkgs.stdenv;
+          name: type: let baseName = baseNameOf (toString name); in ! (
+            # Filter out cabal build products
+            baseName == "dist" ||
+            # Filter out files which don't affect cabal build
+            lib.hasSuffix ".nix" baseName
```

Comment:
Thanks for the review. The goal of these source filters is to avoid nix-build rebuilds of cabal packages when only "irrelevant" files have been changed. For example, the common case of when I edit a `.nix` file within the package's source directory. Try this:

    nix-build -A cardano-sl-wallet-new  # should download from hydra cache
    touch wallet-new/hello
    nix-build -A cardano-sl-wallet-new  # now it wants to rebuild everything

Reply:
Yeah, the purpose is clear. It's just that one wouldn't normally expect a function named “clean haskell source” to pay attention to `.nix` files.

Reply:
OK, now it's called `localLib.cleanSourceTree`.